benchopt/base.py

```diff
@@ -517,3 +518,45 @@ def __reduce__(self):
             self._module_filename, module_hash, self._parameters, dataset,
             str(self._import_ctx._benchmark_dir)
         )
+
+    def default_split(self, *arrays):
+        i_train, i_test = next(self._cv)
+        res = ()
+        for x in arrays:
+            res = (*res, x[i_train], x[i_test])
+        return res
+
+    def custom_split(self, *arrays):
+        if not hasattr(self, "_cv"):
+            # this define _cv only once
+            self._cv = itertools.cycle(self.cv.split(*arrays))
```

Comment:
```suggestion
```

Reply:
maybe you forgot to change something here? I don't see the difference